### DISCUSSION TITLE
Cannot read properties of undefined (reading 'tabs')

### DISCUSSION THREAD

**Discussion by candcconsulting** [iTwin/itwinjs-core#2570] [Q&A] [answered]
I am getting the following error in two apps.

I am trying to upgrade 1, and noticed my panel had gone, but also no tabs are displayed in the viewer, ie treeview

I tried my other app which was working 06/OCT (and indeed the deployed version is still working), but both apps now show this error when running locally.

I will try and rebuild a viewer from scratch.

Frontstage.tsx:302 
        
       Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'tabs')
    at Frontstage.tsx:302
    at Module.produce [as default] (immerClass.ts:94)
    at appendWidgets (Frontstage.tsx:299)
    at addMissingWidgets (Frontstage.tsx:348)
    at fetchFrontstageState (Frontstage.tsx:878)

**Comment by aruniverse** [comment]
Glenn what version of iTwinjs and iTwin viewer are you on?

**Reply by candcconsulting** [reply]
Hi Arun,
    "@bentley/itwin-client": "^2.19.8",
    "@bentley/itwin-viewer-react": "^6.2.0", 
    Is that enough ?

**Reply by aruniverse** [reply]
Glenn, the "@bentley/itwin-viewer-react" has been deprecated in favor of "@itwin/web-viewer-react".
The bentley scoped one youre trying to use is probably incompatible with some of the newer changes in iTwin.js. Should be very simple for you to upgrade; try adding a dep on "@itwin/web-viewer-react" in you app, and replace all imports from the old package to the new one. Let me know if you have any issues with that.

**Reply by candcconsulting** [reply]
Thanks Arun ... Can you post that as a suggested answer rather than a reply so I can mark it as the the answer ...
I had to drop some code that handled missing views in the iModels, but all working again.  Much appreciated

**Reply by aruniverse** [reply]
Glad to help :) Glenn, im curious what version of iTwin.js you were upgrading from.

**Comment by aruniverse** [comment, marked as answer]
Looks like the error is coming from an incompatibility under hood in the @bentley/itwin-viewer-react package with the latest iTwin.js version. 

Heads up, the [@bentley/itwin-viewer-react](https://www.npmjs.com/package/@bentley/itwin-viewer-react) has been deprecated in favor of [@itwin/web-viewer-react](https://www.npmjs.com/package/@itwin/web-viewer-react).

Luckily, it should be very simple for you to upgrade; try adding a dep on "@itwin/web-viewer-react" in your app, and replacing all imports from the old package to the new one.
